Rubberduck.Parsing/SelectionExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Antlr4.Runtime;
+using Rubberduck.VBEditor;
+using System;
+
+namespace Rubberduck.Parsing
+{
+    /// <summary>
+    /// Provide extensions on selections & contexts/tokens
+    /// to assist in validating whether a selection contains
+    /// a given context or a token. 
+    /// </summary>
+    public static class SelectionExtensions
+    {
+        public static bool Contains(this Selection selection, IToken token)
+        {
+            return
+                (((selection.StartLine == token.Line) && (selection.StartColumn - 1) <= token.Column) || (selection.StartLine < token.Line))
+             && (((selection.EndLine == token.Line) && (selection.EndColumn - 1) >= (token.EndColumn())) || (selection.EndLine > token.Line));
+        }
+
+        public static bool Contains(this ParserRuleContext context, Selection selection)
+        {
+            return
+               (((selection.StartLine == context.Start.Line) && (selection.StartColumn - 1) <= context.Start.Column) || (selection.StartLine < context.Start.Line))
+            && (((selection.EndLine == context.Stop.Line) && (selection.EndColumn - 1) >= (context.Stop.EndColumn())) || (selection.EndLine > context.Stop.Line));
```

Comment:
those are pretty long lines... could we split them at each `&&`? again? Same applies above

Reply:
Same concern regarding the end line as above.